alacritty_terminal/src/term/mod.rs

```diff
@@ -1457,6 +1457,57 @@ impl<T> Term<T> {
             rendered: false,
         }
     }
+
+    fn grid_to_string_between(&self, y0: Line, y1: Line) -> String {
+        let grid = &self.grid;
+
+        let mut s = String::with_capacity(*(y1 - y0) * *grid.cols());
+
+        for y in (*y0..*y1).rev() {
+            let row = &grid[y];
+            let wrapline = row.last().unwrap().flags.contains(Flags::WRAPLINE);
+            let mut line_len = 0;
+
+            // Calculate the line length. We need to do this to avoid trailing spaces.
+            if wrapline {
+                line_len = *grid.cols();
+            } else {
+                for n in (0..*grid.cols()).rev() {
+                    if row[Column(n)].c != ' ' {
+                        line_len = n + 1;
+                        break;
+                    }
+                }
+            }
+
+            // Add the chars to the string.
+            for x in 0..line_len {
+                s.push(row[Column(x)].c);
+            }
+
+            if !wrapline {
+                s.push('\n');
+            }
+        }
+
+        // Ensure that the string ends with a newline.
+        if !s.ends_with('\n') {
+            s.push('\n');
+        }
```

Comment:
The last line cannot be `wrapline` iirc, so I don't think this branching is necessary?

Reply:
While that is probably true for the last line in the grid, the last **visible** line can be wrapline if you scroll up.